### PR TITLE
Update tests and GitHub workflow

### PR DESCRIPTION
**Pull request by cbandy**
This fixes the few issues I encountered while trying to run the test suite locally. For the most part, these are small adjustments to the parser spec to match the changes in fec54b52e50e4ba5e4567086d092113c06cf54aa. Occurrences of `qall` are now `cquit`, which I learned exits non-zero.

This also uses MiniTest to run the parser specs during a pull request.

## Summary by Sourcery

Align parser tests, helpers, and documentation with the current coverage data structures and add automated MiniTest execution to the CI workflow.

Enhancements:
- Refine LLVM JSON parser line handling by moving line sorting to occur only when a source file path is present, avoiding unnecessary work for skipped entries.

CI:
- Enhance the test workflow to install MiniTest, check out its repository, and run parser specs under MiniTest alongside the existing headless test runner, updating actions to checkout v4 and ensuring failing tests cause a non-zero exit.

Documentation:
- Update CONTRIBUTING instructions for running the test suite to use the new headless command that exits non-zero on failure.

Tests:
- Update parser and edge-case specs to use the new coverage fields and assertions, adjust temporary fixture content, and refine expectations around path normalization and line ordering.
- Improve the manual headless test runner to validate parsed coverage structures more robustly and to exit non-zero on failure via Neovim commands.
- Introduce shared test helpers that configure luassert, adjust runtime path, and provide sample coverage fixtures using long string syntax.

Chores:
- Add a basic .editorconfig file to standardize editor behavior across the project.

## Summary by CodeRabbit

* **Chores**
  * Added EditorConfig to standardize line endings, final newlines, and indentation.
  * Upgraded CI workflow and test setup for more reliable dependency handling and failure reporting.

* **Tests**
  * Adjusted test runner exit behavior and assertion style; integrated additional assertion tooling and runtime-path setup.
  * Updated test fixtures formatting and expectations to match parser outputs.

* **Refactor**
  * Normalized parser output shape and deferred internal sorting to a common step for consistent results.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,16 +13,32 @@ jobs:
...
-          cd ${{ github.workspace }}
+          nvim --headless -u NONE +"lua dofile('test/run_all.lua')" +cq
+
+          eval "$(luarocks path)"
           nvim --headless -u NONE \
-            +"lua dofile('test/run_all.lua')" +qa!
+          +"set rtp+=$(pwd)/build/mini.test" \
+          +"lua require('mini.test').setup()" \
+          +"lua MiniTest.run_file('test/spec/parsers_spec.lua')" \
+          +"cquit"
```
[the remark this answers is not in the document]
Comment (cbandy) [reply .github/workflows/test.yml:48]: `cquit` is intentional here. MiniTest will exit early with the correct status code, and this status code of 1 indicates that MiniTest did not execute correctly.

**Comment by mr-u0b0dy** [comment]
@cbandy Thank you for the PR. I'll review in sometime and merge this PR.

**Review by mr-u0b0dy — APPROVED** [review]
The PR looks good. It is a nice approach to use cquit and MiniTest.
